### ISSUE TITLE
PTR handling when ipv6addr field present is broken

### ISSUE THREAD

**Issue by howels** [govdelivery/infoblox#43]
I guess our Infoblox has ipv6 enabled and the Gem throws errors when looking for PTR records using ipv4 (which is used primarily).

```
test = Infoblox::Ptr.find(connection, {"ipv4addr" => "10.10.10.10"}).first
Infoblox::Error: Error: 400 { "Error": "AdmConProtoError: Unknown argument/field: 'ipv6addr'",
  "code": "Client.Ibap.Proto",
  "text": "Unknown argument/field: 'ipv6addr'"
}
    from /gems/infoblox-0.5.3/lib/infoblox/connection.rb:89:in `block in wrap'
    from /gems/infoblox-0.5.3/lib/infoblox/connection.rb:87:in `tap'
    from /gems/infoblox-0.5.3/lib/infoblox/connection.rb:87:in `wrap'
    from /gems/infoblox-0.5.3/lib/infoblox/connection.rb:16:in `get'
    from /gems/infoblox-0.5.3/lib/infoblox/resource.rb:110:in `find'
    from (irb):15
    from /ruby-2.3.1/bin/irb:11:in `<main>'
```

**Comment by unclebilly** [comment]
Hi @howels, 
It looks to me like your version of Infoblox (which I'm guessing is <= 1.1) does not support the `ipv6addr` field for the `Ptr` record.  Can you confirm by looking at the WAPI documentation?  If that's the case, I'd entertain a pull request with conditional logic around using that field based on the `WAPI_VERSION` constant.

**Comment by howels** [comment]
Hi @unclebilly,

We have Infoblox version 7.3.5-333643, which includes ipv6addr fields for the Ptr record, however only one of ipv4addr or ipv6addr can exist at a time according to the constructor. So I'm guessing that by searching Ptr records based on an ipv4addr the ipv6addr will never be set?

thanks

`#Constructor for an IPv4 PTR object
 my $bind_ptr = Infoblox::DNS::Record::PTR->new (
         ipv4addr              => $ipv4addr,                                                         # Required
         ptrdname              => $fqdn,                                                             # Required
         comment               => $string,                                                           # Optional / Default is empty
         disable               => "true" | "false"                                                   # Optional / Default "false"
         extattrs              => { $string => $extattr, ... },                                      # Optional / Default is undefined
         extensible_attributes => { $string => $string|$num, $string => [ $string|$num, ...], ... }, # Optional / Default is undefined
         last_discovered       => $string,                                                           # Optional / Default is empty
         name                  => $fqdn,                                                             # Optional
         ttl                   => $num | undef,                                                      # Optional / Default is undefined
         views                 => [$View1],                                                          # Optional / Default is "default" view
         creator               => "DYNAMIC" | "STATIC",                                              # Optional / Default is "STATIC"
         ddns_protected        => "true" | "false",                                                  # Optional / Default is "false"
         ddns_principal        => $string,                                                           # Optional / Default is undefined
         forbid_reclamation    => "true" | "false",                                                  # Optional / Default is "false"
     );
 #Constructor for an IPv6 PTR object
 my $bind_ptr = Infoblox::DNS::Record::PTR->new (
         ipv6addr              => $ipv6addr,                                                         # Required
         ptrdname              => $fqdn,                                                             # Required
         comment               => $string,                                                           # Optional / Default is empty
         disable               => "true" | "false"                                                   # Optional / Default "false"
         extattrs              => { $string => $extattr, ... },                                      # Optional / Default is undefined
         extensible_attributes => { $string => $string|$num, $string => [ $string|$num, ...], ... }, # Optional / Default is undefined
         name                  => $fqdn,                                                             # Optional
         ttl                   => $nun | undef,                                                      # Optional / Default is undefined
         views                 => [$View1],                                                          # Optional / Default is "default" view
         creator               => "DYNAMIC" | "STATIC",                                              # Optional / Default is "STATIC"
         ddns_protected        => "true" | "false",                                                  # Optional / Default is "false"
         ddns_principal        => $string,                                                           # Optional / Default is undefined
         forbid_reclamation    => "true" | "false",                                                  # Optional / Default is "false"
     );
`

**Comment by unclebilly** [comment]
@howels I see - it appears that when requesting an IPv4 PTR record, one should only specify `ipv4addr` in the `_return_fields`, and vice-versa for an IPv6 PTR.  You can work around this problem by specifying the `_return_fields`, which is a comma-separated list of fields that you want to receive: 

    Infoblox::Ptr.find(connection,
      'ipv4addr~': '10.*', '_max_results': 1, '_return_fields': 'ipv4addr,ptrdname,name')

This gem tries to calculate the `_return_fields` automatically but in this case the logic is more complex than the general case.

**Comment by unclebilly** [comment]
As a side note, we are using WAPI version 2.0 and I do not have this problem at all - for instance, this code works fine for me: 

    Infoblox.wapi_version='2.0'
    connection = Infoblox::Connection.new(...)
    Infoblox::Ptr.find(connection, 'ipv4addr~' => '10.*', '_max_results' => 100).map(&:ipv6addr)

**Comment by unclebilly** [comment]
I released v1.0.1 just now, which should solve this problem without using the workaround above.  I'm going to release v2.0.0 shortly, which makes WAPI version 2 the default.  Might want to wait for that gem if you're interested in avoiding other strange bugs like this 😸
